src/nickserv.c

```diff
@@ -708,7 +708,7 @@ static const char *
 visible_email_addr(struct userNode *user, struct handle_info *hi)
 {
     if (hi->email_addr) {
-        if (oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
+        if (hi == user->handle_info || oper_has_access(user, nickserv, nickserv_conf.email_visible_level, 1)) {
```

Comment:
Why do users need to see their own email address?  The reason this was not allowed in the first place was to prevent information leakage in case someone's account gets compromised, and a malicious user is able to log into someone else's account.